frontend/src/modules/2DViewer/view/components/LayersWrapper.tsx

```diff
@@ -0,0 +1,154 @@
+import React from "react";
+
+import { View as DeckGlView } from "@deck.gl/core";
+import { ViewContext } from "@framework/ModuleContext";
+import { useViewStatusWriter } from "@framework/StatusWriter";
+import { PendingWrapper } from "@lib/components/PendingWrapper";
+import { useElementSize } from "@lib/hooks/useElementSize";
+import { Rect2D, outerRectContainsInnerRect } from "@lib/utils/geometry";
+import { Interfaces } from "@modules/2DViewer/interfaces";
+import { LayerManager, LayerManagerTopic } from "@modules/2DViewer/layers/LayerManager";
+import { usePublishSubscribeTopicValue } from "@modules/2DViewer/layers/delegates/PublishSubscribeDelegate";
+import { BoundingBox } from "@modules/2DViewer/layers/interfaces";
+import { PreferredViewLayout } from "@modules/2DViewer/types";
+import { ColorLegendsContainer } from "@modules/_shared/components/ColorLegendsContainer";
+import { ColorScaleWithId } from "@modules/_shared/components/ColorLegendsContainer/colorLegendsContainer";
+import { ViewportType } from "@webviz/subsurface-viewer";
+import { ViewsType } from "@webviz/subsurface-viewer/dist/SubsurfaceViewer";
+
+import { ReadoutWrapper } from "./ReadoutWrapper";
+
+import { PlaceholderLayer } from "../customDeckGlLayers/PlaceholderLayer";
+import { DeckGlLayerWithPosition, recursivelyMakeViewsAndLayers } from "../utils/makeViewsAndLayers";
+
+export type LayersWrapperProps = {
+    layerManager: LayerManager;
+    preferredViewLayout: PreferredViewLayout;
+    viewContext: ViewContext<Interfaces>;
+};
+
+export function LayersWrapper(props: LayersWrapperProps): React.ReactNode {
+    const [prevBoundingBox, setPrevBoundingBox] = React.useState<BoundingBox | null>(null);
+
+    const mainDivRef = React.useRef<HTMLDivElement>(null);
+    const mainDivSize = useElementSize(mainDivRef);
+    const statusWriter = useViewStatusWriter(props.viewContext);
+
+    usePublishSubscribeTopicValue(props.layerManager, LayerManagerTopic.LAYER_DATA_REVISION);
+
+    const viewports: ViewportType[] = [];
+    const viewerLayers: DeckGlLayerWithPosition[] = [];
+    const viewportAnnotations: React.ReactNode[] = [];
+    const globalColorScales: ColorScaleWithId[] = [];
+
+    const views: ViewsType = {
+        layout: [1, 1],
+        viewports: viewports,
+        showLabel: false,
+    };
+
+    let numCols = 0;
+    let numRows = 0;
+
+    let numLoadingLayers = 0;
+
+    const viewsAndLayers = recursivelyMakeViewsAndLayers(props.layerManager);
```

Comment:
`recursivelyMakeViewsAndLayers()` is used inside itself - having the `LayerManager` as an input argument would break that. I could change it to take a `GroupDelegate` as argument.
```
recursivelyMakeViewsAndLayers(
    groupDelegate: GroupDelegate,
    numCollectedLayers: number = 0
)
```

Reply:
Good point! Works fine for me 👍